fairlearn/reductions/_exponentiated_gradient/_lagrangian.py

```diff
@@ -148,9 +148,14 @@ def _call_oracle(self, lambda_vec):
         signed_weights = self.obj.signed_weights() + self.constraints.signed_weights(lambda_vec)
         if isinstance(self.constraints, ClassificationMoment):
             redY = 1 * (signed_weights > 0)
+            redW = signed_weights.abs()
+        elif isinstance(self.constraints,CDF_DemographicParity):
+            redY = self.constraints.optimal_label(signed_weights)
+            redW = pd.Series(1,index=redY.index)
         else:
             redY = self.y
```

Comment:
please leave a comment there on what cases would fall into the else branch (I assume that's BGL?)